src/watch.rs

```diff
@@ -73,6 +86,20 @@ fn update_watcher(
     Ok(())
 }
 
+fn unwatch_all(watcher: &mut RecommendedWatcher, old_files: &HashSet<PathBuf>) -> Fallible {
+    for path in old_files {
+        watcher.unwatch(&path)?;
```

Comment:
This is probably a rather costly operation as it involves multiple system calls. A more lightweight approach would probably be to share an `suspend: Arc<AtomicBool>` with each background thread which leads to all events being ignored when `suspend.load(Ordering::SeqCst)` is true.

Reply:
I will give it a try, but not sure if this works with the *debounce* watcher that delays by 1 second.

Reply:
With an implementation as in #165, it seems not to work. Maybe I missed something, but I think it is due to the delay in debounce watcher.

Reply:
(As an aside, do you really need the debounced watcher as you already coalesce events anyway? You might want to wait for quiescence though, i.e. reset the timeout whenever an event arrives as there is little use to rebuilding while the files still change.)